### ISSUE TITLE
Logging middleware before Transactional

### ISSUE THREAD

**Issue by acasademont** [SimpleBus/SymfonyBridge#18]
Hi Matthias!

I've been playing these week with the shiny new version and everything works like a charm, thanks! However, I was wondering why the Transactional Doctrine middleware is registered before the Logging middleware, I would prefer the transaction to be "as close as possible" to the real work done inside the command. If somehow the logging failed that would mean rolling back the transaction too, which is not desirable IMHO

> [2015-05-29 12:11:21] doctrine.DEBUG: "START TRANSACTION" [] []
> [2015-05-29 12:11:21] command_bus.DEBUG: Started handling a message {"message":"[object] (Ulabox\\Inventory\\Application\\Command\\Sku\\LinkWithProduct: {})"} []

```
array(5) {
  [0] =>
  string(71) "simple_bus.command_bus.finishes_command_before_handling_next_middleware"
  [1] =>
  string(56) "simple_bus.event_bus.handles_recorded_mesages_middleware"
  [2] =>
  string(64) "simple_bus.doctrine_orm_bridge.wraps_next_command_in_transaction"
  [3] =>
  string(41) "simple_bus.command_bus.logging_middleware"
  [4] =>
  string(62) "simple_bus.command_bus.delegates_to_message_handler_middleware"
}
```

Thanks!!

**Comment by matthiasnoback** [comment]
You are totally right! Sounds like something I overlooked. Not sure if it's
safe to change the priorities though. That might require a new major
version.

In your case you can for now just disable logging and register the
middleware yourself of course.
Op 29 mei 2015 12:28 schreef "Albert Casademont" <notifications@github.com>:

**Comment by acasademont** [comment]
Yes I'll do that. Digging a bit further into the problem it seems that the logger middleware gets a -999 priority, just one above the handling middleware that gets a -1000. The Transactional gets a 100, so it's definitely above it!

**Comment by acasademont** [comment]
I guess that what you really wanted is to set the priority of the logging middleware to 999, not -999 just after the `FinishesHandlingMessageBeforeHandlingNext`

**Comment by matthiasnoback** [comment]
@acasademont You are completely right. 

At first I thought this would be a bad change since it might be considered a BC break. However, it's logging, and I think it's reasonable to assume that it no-one would rely on the logging middleware to have a specific priority. Anyway, now it's more like a bug :) So we should fix it.
